### PR TITLE
Adding a managed spot training example

### PR DESCRIPTION
**Pull request by bbalaji-ucsd**
*Issue #, if available:*

*Description of changes:*
Added a new exampled that uses managed spot training for the managed spot training example. 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: reinforcement_learning/rl_managed_spot_cartpole_coach/rl_managed_spot_cartpole_coach_gymEnv.ipynb [hunk 1]
```diff
@@ -0,0 +1,581 @@
...
+    "estimator = RLEstimator(entry_point=\"train-coach.py\",\n",
+    "                        source_dir='src',\n",
+    "                        dependencies=[\"common/sagemaker_rl\"],\n",
+    "                        toolkit=RLToolkit.COACH,\n",
+    "                        toolkit_version='0.11.0',\n",
+    "                        framework=RLFramework.MXNET,\n",
+    "                        role=role,\n",
+    "                        train_instance_type=instance_type,\n",
+    "                        train_instance_count=1,\n",
+    "                        train_use_spot_instances=True,\n",
```
Comment (annaluo676) [inline reinforcement_learning/rl_managed_spot_cartpole_coach/rl_managed_spot_cartpole_coach_gymEnv.ipynb]: A note to myself -- enabling spot instance

File: reinforcement_learning/rl_managed_spot_cartpole_coach/README.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
+# Simple Cartpole Example
+
+## What is cartpole?
+
+This is a classic example of a problem that reinforcement learning can solve.  It's a simulation of balancing a broom upright by balancing it on your hand.  The broom is the "pole" and your hand is replaced with a "cart" moving back and forth on a linear track.  This simplified example works in 2 dimensions, so the cart can only move in a line back and forth, and the pole can only fall forwards or backwards, not to the sides.
+
+## This Example
+
+Sample Cart Pole Example using SageMaker RL with base docker image containing Coach, MxNet and OpenAI Gym. This is a toy example taken from 
+[Coach Quick Start Guide](https://github.com/NervanaSystems/coach/blob/master/tutorials/0.%20Quick%20Start%20Guide.ipynb). It demonstrates how you can use the `RLEstimator` from the SageMaker Python SDK in `script` mode.
```
Comment (annaluo676) [inline reinforcement_learning/rl_managed_spot_cartpole_coach/README.md]: Maybe highlight the spot instance by adding something like 'This notebook tackles the exact same problem with the same solution, but it has been modified to be able to run using SageMaker Managed Spot infrastructure. SageMaker Managed Spot uses [EC2 Spot Instances]'

File: reinforcement_learning/rl_managed_spot_cartpole_coach/rl_managed_spot_cartpole_coach_gymEnv.ipynb [hunk 3]
```diff
@@ -0,0 +1,581 @@
...
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Cart-pole Balancing Model with Amazon SageMaker on SageMaker Managed Spot Training\n",
+    "\n",
+    "The example here is almost the same as [Cart-pole Balancing Model with Amazon SageMaker](https://github.com/awslabs/amazon-sagemaker-examples/blob/master/reinforcement_learning/rl_cartpole_coach/rl_cartpole_coach_gymEnv.ipynb).\n",
+    "\n",
+    "This notebook tackles the exact same problem with the same solution, but it has been modified to be able to run using SageMaker Managed Spot infrastructure. SageMaker Managed Spot uses [EC2 Spot Instances](https://aws.amazon.com/ec2/spot/) to run Training at a lower cost.\n",
```
Comment (annaluo676) [inline reinforcement_learning/rl_managed_spot_cartpole_coach/rl_managed_spot_cartpole_coach_gymEnv.ipynb]: highlight the difference compared to the other cartpole example

**Comment by bbalaji-ucsd** [comment]
I've addressed the comments raised by Anna.

**Comment by annaluo676** [comment]
Looks good.

**Pull request merged**